### PR TITLE
Refactor LS package loader

### LINKED ISSUE #41940: [Improvement]: Fix delay in code completions during VSCode startup.
### Description

Currently there is a noticeable delay in completions when VSCode is starting up which impacts the overall user experience. Hence, we should come up with a proper solution to optimize the initial performance.

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by mindula**
## Purpose
$title.
PR for master - #42173

Fixes #41940

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by KavinduZoysa — APPROVED** [review]

**Comment by KavinduZoysa** [comment]
Let's work on disabled tests separately.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSPackageLoader.java [hunk 1]
```diff
@@ -70,69 +97,156 @@ public static LSPackageLoader getInstance(LanguageServerContext context) {
...
+            notificationTaskId = taskId;
+            Map<String, ModuleInfo> packagesList = new HashMap<>();
+            CompletableFuture.runAsync(() -> {
+                this.languageClient = context.get(ExtendedLanguageClient.class);
+                this.languageClient.showMessage(
+                        new MessageParams(MessageType.Info, "Indexing Ballerina packages: " +
+                                "some completions may not be available until indexing is complete."));
+                if (languageClient == null) {
+                    return;
+                }
```
Comment (nipunayf) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSPackageLoader.java:127]: In this case, shouldn't the null check be before the invocation of the show message?

Comment (mindula) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSPackageLoader.java:127]: Yes my mistake

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/imports/AddModuleToBallerinaTomlCodeAction.java [hunk 2]
```diff
@@ -132,10 +130,9 @@ public String getName() {
     
     private List<PackageVersion> getAvailablePackageVersionsFromLocalRepo(
             LSPackageLoader lsPackageLoader, Project project, String orgName, String pkgName) {
-        BallerinaUserHome ballerinaUserHome = 
-                BallerinaUserHome.from(project.projectEnvironmentContext().environment());
-        PackageRepository localRepository = ballerinaUserHome.localPackageRepository();
-        List<ModuleInfo> modules = lsPackageLoader.getLocalRepoPackages(localRepository);
+//        BallerinaUserHome ballerinaUserHome =
+//                BallerinaUserHome.from(project.projectEnvironmentContext().environment());
```
Comment (KavinduZoysa) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/imports/AddModuleToBallerinaTomlCodeAction.java:134]: Let's delete this

File: tests/language-server-integration-tests/src/test/java/org/ballerinalang/langserver/test/completion/CompilerPluginCompletionTests.java [hunk 3]
```diff
@@ -35,7 +35,7 @@ public void compilePlugins() {
     }
 
     @Override
-    @Test(dataProvider = "completion-data-provider")
+    @Test(dataProvider = "completion-data-provider", enabled = false)
```
Comment (KavinduZoysa) [inline tests/language-server-integration-tests/src/test/java/org/ballerinalang/langserver/test/completion/CompilerPluginCompletionTests.java:38]: Shall we create an issue for this?

Comment (mindula) [reply tests/language-server-integration-tests/src/test/java/org/ballerinalang/langserver/test/completion/CompilerPluginCompletionTests.java:38]: Created an issue https://github.com/ballerina-platform/ballerina-lang/issues/42226

**Comment by nipunayf** [comment]
Closing this PR as these changes are not needed for the `worker_change` branch.

**Pull request closed without merging**